src/picongpu/include/plugins/common/txtFileHandling.hpp

```diff
@@ -46,28 +47,46 @@ using namespace boost::filesystem;
     bool restoreTxtFile( std::ofstream& outFile, std::string filename,
                          uint32_t restartStep, const std::string restartDirectory )
     {
-        if( outFile.is_open() )
-            outFile.close();
-
+        /* get restart time step as string */
         std::stringstream sStep;
         sStep << restartStep;
 
+        /* set location of restart file and output file */
         path src( restartDirectory + std::string("/") + filename +
                   std::string(".") + sStep.str() );
         path dst( filename );
 
-        copy_file( src,
-                   dst,
-                   copy_option::overwrite_if_exists );
-
-        outFile.open( filename.c_str(), std::ofstream::out | std::ostream::app );
-        if( !outFile )
+        /* check whether restart file exists */
+        std::ifstream restartFile(src.c_str());
```

Comment:
That's an interesting way of checking that.
We usually use `if (!boost::filesystem::exists(src))` for that - maybe that's more suited here, too.